Work through release process with docs (#1213)
collective/icalendar: docs/contribute/maintenance.rst

Reviewer: ```suggestion
#.  Create a tag for the release on its release branch ``*.x`` and see if the `CI tests`_ are running.
```

Reviewer: @niccokunzmann I realized that for every stable release, we toggle this from `True` to `False`, then back to `True` on `main`. I think that's extra work that could be avoided. I'll try to explain in the overall review comment.

Author: What I really would like to do is to run one command and then, the version is released, all edits done automatically. I think the changelog issue you suggested can help with that.

.. seealso:: `About code owners <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners>`_

Release Glossary
----------------

major release:
    Increase of the first version number. E.g. ``7.0.0`` to ``8.0.0``
minor release:
    Increase of the second version number. E.g. ``7.0.0`` to ``7.1.0``
patch release:
    Increase of the third version number. E.g. ``7.0.0`` to ``7.0.1``
stable release:
    A release with three numbers divided by two dots. E.g. ``7.0.0``, ``6.3.12``
unstable release:

>     A release of an alpha or beta version. E.g. ``7.0.0a1``

Author: Right! This is in the changelog, too. Maybe there should be one place for this?

    .. note::

        The remaining steps may be performed after the release because they pertain exclusively to documentation, which isn't included in the release.
        The following examples were used for the 7.0.0 release.

>         Making a pull request won't effect the version switcher on Read the Docs until it gets on the ``main`` branch.

Reviewer: No longer relevant.

```suggestion
```

>         Making a pull request won't have any effect to the version switcher on Read the Docs until it gets on to the ``main`` branch.

Reviewer: ```suggestion
        Making a pull request won't effect the version switcher on Read the Docs until it gets on the ``main`` branch.
```

    .. important::

> 

Reviewer: No longer relevant.

```suggestion
```

        Edit and push directly on the ``main`` branch after these steps.

        .. code-block: shell

            git checkout main
            git pull

>     #.  When cutting a new *major release* version, on the ``main`` branch, update :file:`docs/_static/version-switcher.json` to match that version.

Reviewer: ```suggestion
    #.  When cutting a new *major release* version, checkout the ``main`` branch, and update :file:`docs/_static/version-switcher.json` to match that version.
    
        ..  code-block:: shell
        
            git checkout main
            git pull
```

>         #.  Copy the second previous major version and renumber it to the first previous version, in other words, copy ``5.x`` and renumber the copy to ``6.x``.

Reviewer: ```suggestion
        #.  Copy the second previous major version stanza and renumber it to the first previous version.
            In other words, duplicate the ``5.x`` stanza, and renumber the copy to ``6.x``.
```

                    "url": "https://icalendar.readthedocs.io/en/stable/",
                    "preferred": "true"
                },
    
        #.  On the previous numbered major release branch (e.g. ``6.x`` if you release ``7.x`` now), show the warning banner.

            .. code-block:: python
    
                html_theme_options = {
                    # ...
                    "show_version_warning_banner": True,

>             .. code-block: shell

Reviewer: ```suggestion
            .. code-block:: shell
```

    
                git checkout 7.x

>                 git pull

Reviewer: It was still there. Not sure what you were looking at. I'll commit this and 4 other suggestions that got lost.

>     #.  When cutting a *minor or patch release* version, on the ``main`` banch, update :file:`docs/_static/version-switcher.json` to match that version's tag name.

Reviewer: ```suggestion
    #.  When cutting a *minor or patch release* version, on the ``main`` branch, update :file:`docs/_static/version-switcher.json` to match that version's tag name.
```

#.  Configure `Read the Docs <https://app.readthedocs.org/projects/icalendar/>`_.

    #.  After an *unstable release*:

>         Deactivate any unstable releases.

Reviewer: This will be true. I think that we can automate it in RTD, but let's keep it manual for now, until we fully understand how it activates branches.

```suggestion
    #.  After a *minor release*, *patch release*, or an *unstable release*, deactivate previous releases in that release line.
```

                "show_version_warning_banner": True,

    #.  On the previous numbered major release branch, show the warning banner.
        For example, when releasing 7.0.0, checkout ``6.x``, and update it as shown.

        .. code-block:: python
    
            html_theme_options = {
                # ...
                "show_version_warning_banner": True,

#.  Configure `Read the Docs <https://app.readthedocs.org/projects/icalendar/>`_.

    #.  After a *minor release*, *patch release*, or an *unstable release*, deactivate previous releases in that release line.
        Click on the ellipsis icon, and select :guilabel:`Configure version`.
        Toggle the :guilabel:`Active` switch to deactivate the version.

    #.  After a *major release*, activate its version.
        Click `Add version <https://app.readthedocs.org/dashboard/icalendar/version/create/>`_.
        Search for the new major version ``#.x``.
        Click on the version that appears in the select menu that matches your search.

>         Toggle the :guilabel:`Active` switch to activate the version.

Reviewer: This is actually handled automatically, but we should verify.

```suggestion
    #.  After a *major release*, verify that the version was activated automatically on Read the Docs.
```
